test/baselines/adapter_TS_React/admin/src/components/settings.tsx

```diff
@@ -1,124 +1,136 @@
-	// 1. store every reference to a checkbox in a variable
-	// private chkWriteLogFile: HTMLInputElement | null | undefined;
-	// 2. add the click event handler in `componentDidMount`
-	// if (this.chkWriteLogFile != null) {
-	// 	$(this.chkWriteLogFile).on("click", this.handleChange as any);
-	// }
-	// 3. remove the click handler in `componentWillUnmount`
-	// if (this.chkWriteLogFile != null) {
-	// 	$(this.chkWriteLogFile).off("click", this.handleChange as any);
-	// }
-
-	private parseChangedSetting(target: HTMLInputElement | HTMLSelectElement): unknown {
-		// Checkboxes in MaterializeCSS are messed up, so we attach our own handler
-		// However that one gets called before the underlying checkbox is actually updated,
-		// so we need to invert the checked value here
-		return target.type === "checkbox"
-			? !(target as any).checked
-			: target.type === "number"
-			? parseInt(target.value, 10)
-			: target.value;
-	}
-
-	// gets called when the form elements are changed by the user
-	private handleChange(event: React.FormEvent<HTMLElement>): boolean {
-		const target = event.target as HTMLInputElement | HTMLSelectElement; // TODO: more types
-		const value = this.parseChangedSetting(target);
-		return this.updateSettings(target.id, value);
-	}
-
-	private updateSettings(setting: string, value: unknown): boolean {
-		// store the setting
-		this.putSetting(setting, value, () => {
-			// and notify the admin UI about changes
-			this.props.onChange(composeObject(entries(this.state).filter(([k]) => !k.startsWith("_"))));
-		});
-		return false;
-	}
 
-	/**
-	 * Reads a setting from the state object and transforms the value into the correct format
-	 * If no value is set, the default value is returned
-	 * @param key The setting key to lookup
-	 */
-	private getSetting(key: string, defaultValue?: unknown): unknown {
-		const ret = this.state[key];
-		return ret != undefined ? ret : defaultValue;
+		this.state = {};
 	}
 
-	/**
-	 * Saves a setting in the state object and transforms the value into the correct format
-	 * @param key The setting key to store at
-	 */
-	private putSetting(key: string, value: unknown, callback?: () => void): void {
-		this.setState({ [key]: value }, callback);
-	}
-
-	public componentDidMount(): void {
-		// update floating labels in materialize design
-		M.updateTextFields();
-
-		// Add checkbox click handlers here
+	renderInput(title: string, attr: string, type: string): React.ReactNode {
+		return (
+			<TextField
+				label={I18n.t(title)}
+				className={this.props.classes.input + " " + this.props.classes.controlElement}
+				value={this.props.native[attr]}
+				type={type || "text"}
+				onChange={(e) => this.props.onChange(attr, e.target.value)}
+				margin="normal"
+			/>
+		);
 	}
 
-	public componentWillUnmount(): void {
-		// Remove checkbox click handlers here
+	renderSelect(
+		title: string,
+		attr: string,
+		options: { value: string; title: string }[],
+		style?: any,
+	): React.ReactNode {
```

Comment:
These methods would be perfect for functional components (and more self-descriptive at the usage site). Probably something for a followup PR though.

Reply:
Agreed: PR and perhaps general discussion with Bluefox about the way forward.